### ISSUE TITLE
SelectOneMenu: Editable displays value instead of label

### ISSUE THREAD

**Issue by JHookDev** [primefaces/primefaces#3477]
## 1) Environment
 - PrimeFaces version: 6.2
 - Does it work on the newest released PrimeFaces version? No Version? 6.2
 - Does it work on the newest sources in GitHub? (Build by source -> https://github.com/primefaces/primefaces/wiki/Building-From-Source)
 - Application server + version: Tomcat 7.0.23
 - Affected browsers: Tested with Chrome and IE
 
 ## 2) Expected behavior
 When rendering the dropdown, if the value matches a value in the select items, it should display the matching label for that value.
 
 
 ## 3) Actual behavior
 When rendering the dropdown, the value is displayed even when the value matches an item in the select items.
 
 ## 4) Steps to reproduce
 Using the sample below XHTML and Bean below, select a value from the dropdown and press submit. Upon the page being rerendered you will see the value for the selected item rather than the label.
  
 
 ## 5) Sample XHTML
 ```
<f:view>
  <h:form id="selectOneMenuForm">
    <p:outputLabel for="editableDropdown" value="Editable: " />
    <p:selectOneMenu id="editableDropdown" value="#{selectOneMenu.selection}"
        effect="fold" editable="true">
        <f:selectItem itemLabel="Select One" itemValue="" />
        <f:selectItems value="#{selectOneMenu.options}" />
    </p:selectOneMenu>
    <p:commandButton id="submit" value="Submit" ajax="false"/>
  </h:form>
</f:view>
```
 
 
 ## 6) Sample bean
 ```
@ManagedBean
public class SelectOneMenu {
	
	private String selection;
    private Map<String,String> options = new HashMap<String, String>();

    @PostConstruct
	public void init() {
		options = new HashMap<String, String>();
		options.put("Label1", "Value1");
		options.put("Label2", "Value2");
	}

	public String getSelection() {
		return selection;
	}

	public void setSelection(String selection) {
		this.selection = selection;
	}

	public Map<String, String> getOptions() {
		return options;
	}

	public void setOptions(Map<String, String> options) {
		this.options = options;
	}

}
```

**Comment by melloware** [comment]
This was fixed before so there must be a regression somewhere:  https://github.com/primefaces/primefaces/issues/2862

**Comment by melloware** [comment]
Can you review this commit and try and determine why its not working?  or is your scenario possibly different?  https://github.com/primefaces/primefaces/commit/1cc5fbffda3e894154320ff7757028957b0d9a16

**Comment by JHookDev** [comment]
@melloware The commit you referenced appears to correct a different issue that was occurring during the decode to set the value on the bean from the component. The issue here is related to when rendering the editable dropdown from the bean value to the component, where the value is displayed when it should be the label when the value matches a select item value. I'll review the source later today if time permits.

**Comment by JHookDev** [comment]
@melloware Found that the issue is in SelectOneMenuRenderer.encodeLabel. When it's editable it uses ComponentUtils.getValueToRender to set the value for the input tag being rendered. One way to fix this would be to cycle through the select items, if any of the values match the valueToRender then set valueToRender to the select item label. In addition to looking for the match, the first line in the example below was moved from the second line of the method since it's only needed for this section of code.

```
...
String valueToRender = ComponentUtils.getValueToRender(context, menu);
if (valueToRender != null) {
    for (SelectItem anItem : selectItems) {
        if (isSelected(context, menu, valueToRender, anItem.getValue().toString(), null)) {
            valueToRender = anItem.getLabel();
        }
    }
    writer.writeAttribute("value", valueToRender, null);
}
...
```

**Comment by tandraschko** [comment]
Thanks - also improved it a bit. Feel free to review.

**Comment by JHookDev** [comment]
Thanks @tandraschko! I see this has been tagged for 6.3. I know that you might not be the person that controls the Elite releases, but do you happen to know if it will be included in a 6.2 Elite release?

**Comment by tandraschko** [comment]
thats up to @mertsincan

**Comment by mertsincan** [comment]
Fixed for 6.2.10.

**Comment by JHookDev** [comment]
Thank you for the update @mertsincan
